api/v1alpha1/envoygateway_types.go

```diff
@@ -91,6 +91,23 @@ type EnvoyGatewaySpec struct {
 	ExtensionAPIs *ExtensionAPISettings `json:"extensionApis,omitempty"`
 }
 
+// KubernetesClientRateLimit defines the rate limit settings for the Kubernetes client.
+type KubernetesClientRateLimit struct {
+	// QPS defines the queries per second (QPS) limit for the Kubernetes client.
+	// If unspecified, defaults to 50. Min value must be greater than or equal to 1.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	QPS float32 `json:"qps,omitempty"`
```

Comment:
I think that we avoided floats so far in the API, as mentioned here: https://github.com/envoyproxy/gateway/issues/5089. 

In this case, is it really meaningful to use a float over an int?